numpy/random/_generator.pyx

```diff
@@ -3537,10 +3537,9 @@ cdef class Generator:
             # approximately zero or when the covariance is not positive-semidefinite
             _factor = u * np.sqrt(abs(s))
         else:
-            _factor = np.sqrt(s)[:, None] * vh
+            _factor = u * np.sqrt(s)
 
-        x = np.dot(x, _factor)
-        x += mean
+        x = mean + np.einsum("ij,...j->...i", _factor, x)
         x.shape = tuple(final_shape)
```

Comment:
BLAS will be faster than einsum. `x` will always be 2D at this point due to previous reshape and transpose is cheap, so just `x @ _factor.T` should be fine and at least keeps all the transposes in one place while keeping the results the same for the 'svd' case. The output shape is what make things tricky. On could transpose `x` in the reshape and do `(_factor @ x).T` but that doesn't seem much of an improvement.

Reply:
Ah I see I thought `x` here could still have a bunch of leading dims. That makes things easier.